### ISSUE TITLE
[localhost] is currently unable to handle this request - Error 500

### ISSUE THREAD

**Issue by bensilver95** [mediagis/nominatim-docker#124]
I recognize this might be a difficult thing to debug, but I've spent a few days poring over this and can't seem to fix it. I very closely followed the instructions for 3.4 and completed steps 1-4. 

The only difference between what I did and the instructions is that, in step 4, the /var/lib/postgresql folder was 10, not 11.

When I visit http://[my-server-ip]:7070, I get HTTP error 500, and the message is very vague: "The page isn't working. [my-ip] is currently unable to handle this request."

I know permissions is a common problem here, so I did chmod -R 777 on the nominatim-docker folder, nominatimdata folder, and /var/lib/postgresql. But it didn't fix the issue. 

Any other common problems that might be causing this message?

Thanks!

**Comment by mtmail** [comment]
The error is probably Docker proxying the :7070 request to Apache running on :8080 and getting no response, and thus no further information to print.

PHP errors are written to `/var/log/apache2/error.log` Does the file contain any hints?

Does running a geocoding query with
```
docker exec -it nominatim sudo -u postgres ./src/build/utils/query.php
```
work? That run the PHP directly and would print errors to the terminal.

**Comment by bensilver95** [comment]
Okay so when I run the above query I receive this error: 
```
PHP Warning:  require_once(/app/src/build/settings/local.php): failed to open stream: Permission denied in /app/src/build/settings/settings.php on line 5
PHP Fatal error:  require_once(): Failed opening required '/app/src/build/settings/local.php' (include_path='.:/usr/share/php') in /app/src/build/settings/settings.php on line 5
```
I tried doing chmod -R 777 on /app/src/build, but it said that directory didn't exist.

The log file didn't really contain much, except for an error for the command I just ran above. It says:
```
[Thu Jun 18 06:25:01.714202 2020] [mpm_prefork:notice] [pid 170971] AH00163: Apache/2.4.29 (Ubuntu) configured -- resuming normal operations
[Thu Jun 18 06:25:01.714234 2020] [core:notice] [pid 170971] AH00094: Command line: '/usr/sbin/apache2'
```

**Comment by mtmail** [comment]
I think `/app/src/build` exists because it's listed in the error message. settings.php checks if the file local.php exist and only then calls require_once on local.php so local.php also exists at that path.

When I tested it locally (not inside Docker) Nominatim works fine if local.php is missing and if it's empty. I get the same error when the file can't be read (I did `chmod 000 local.php` to test).

Experiment more with `chmod`/`chown`/`chgrp`. I think 777 is too much for a service that potentially runs on the internet but for debugging it's fine.

**Comment by bensilver95** [comment]
Am I misunderstanding something about where /app/src/build should exist? I'm not super familiar with the way docker works. But when I simply do "ls /app" it says "cannot access '/app': no such file or directory"

**Comment by bensilver95** [comment]
Okay, I think I figured it out, but I now have a different error. If I run
```
sudo docker exec -it nominatim chmod -R 775 /app/
```
then when I run the geocoding query you suggested above, I no longer get the permissions error, but instead get 
```
PHP Fatal error:  Uncaught Nominatim\DatabaseError: [500]: Failed to establish database connection:SQLSTATE[08006] [7] FATAL:  database "nominatim" does not exist
Is the server running locally and accepting
	connections on Unix domain socket "/var/run/postgresql/.s.PGSQL.5432"?

  thrown in /app/src/lib/DB.php on line 36
```
I saw there was a similar issue open and closed last year, but I double checked that my /var/lib/postgresql path was right, and it is (I realized I should be doing 11, not 10). I also tried the solution suggested [here](https://www.digitalocean.com/community/questions/is-the-server-running-locally-and-accepting-connections-on-unix-domain-socket-var-run-postgresql-s-pgsql-5432) but that didn't work either.

Should I close this issue and open a new one?

**Comment by mtmail** [comment]
One step closer at least.

Now the question is why the postgresql server is not running. Could indeed be related to the 11 vs 10 changed. For example are the postgresql configuration files in /etc/postgresql/11/main/postgresql.conf ? In other projects I spent hours dealing with tiny differences when two versions of postgresql are installed, from header files, paths to binaries, where configuration files are located, which default port is chosen. If your import was small/fast it might be easier to start fresh with the Nominatim 3.5 version https://github.com/mediagis/nominatim-docker/tree/master/3.5 that comes with postgresql 12.

**Comment by bensilver95** [comment]
I realized earlier when I said the path was 10, I wasn't looking within the Docker container, the 10 was from another time I had installed postgresql locally. Within docker, the path is 11, and there is indeed a postgresql.conf file in the path you mentioned, and in the "file locations" section, all the file paths use 11.

I also realized there was a slight typo in the error I pasted in my last comment. If I execute step 4 with an incorrect path (10 instead of 11 for example) the error I get when running the geocoding query is
```
PHP Fatal error:  Uncaught Nominatim\DatabaseError: [500]: Failed to establish database connection:SQLSTATE[08006] [7] FATAL:  database "nominatim" does not exist

  thrown in /app/src/lib/DB.php on line 36
```
but if I have the correct path, I get the following:
```
PHP Fatal error:  Uncaught Nominatim\DatabaseError: [500]: Failed to establish database connection:SQLSTATE[08006] [7] could not connect to server: No such file or directory
	Is the server running locally and accepting
	connections on Unix domain socket "/var/run/postgresql/.s.PGSQL.5432"?

  thrown in /app/src/lib/DB.php on line 36
```
Sorry about that, just a pasting error. So I'm very confident that the problem, once I fixed the path issue, is no long a problem of not being able to find the database nominatim, but now is a problem of not being able to connect to the postgresql server. Is there something I can do to test whether it's running or manually start it?

The import took quite a while, so if possible I'd like to avoid starting over, at least for now.
